File: Domain/DailyMessingItem.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Threading.Tasks;

namespace MessingSystem.Domain
{
    public class DailyMessingItem
    {
        [Key]
        public int Id { get; set; }

        public int DailyMessingId { get; set; }
        public int ItemType { get; set; }

        public decimal Quantity { get; set; }
    }
}

File: wwwroot/js/models/dailyMessingItem.js
var DailyMessingItem = function (data) {
    var self = this;
    if (data) {
        ko.mapping.fromJS(data, {}, self);
    }
    else {
        self.itemType = ko.observable(0);
        self.quantity = ko.observable(0);
    }
}